shortfin/python/shortfin_apps/sd/components/builders.py

```diff
@@ -1,7 +1,9 @@
 from iree.build import *
+from iree.build.executor import FileNamespace
```

Comment:
As a follow-up, can we have a separate tree for compilation? Move this out from under sd/components. Maybe into sd/compile. May want to do more later, but it's generally idea to have a strong separation between compilation and runtime.

Reply:
I was expecting compilation to live in sharktank, like here: https://github.com/nod-ai/SHARK-Platform/tree/main/sharktank/sharktank/models , where we already depend on `torch`, `iree-compiler`, and other packages.

Does SDXL use anything from sharktank? Just the punet model? I put some basic setup docs in the readme that tried to explain the separation: https://github.com/nod-ai/SHARK-Platform?tab=readme-ov-file#support-matrix

I guess shortfin/ makes sense for models that don't use any custom "model recipes"? For example onnx models that just want to use the build system + standard compiler flows.

Reply:
When we do builders with exports enabled it will make much more sense. Currently, we get along without the sharktank import for SDXL, and I think it's prudent to make the cut this way until we have exports prepared with iree.build (those would warrant a spot in sharktank, and their enablement would require the sharktank dep.)
At that point, they can function largely the same but be equipped with export capabilities, and the server can decide which builder to use based on which dependencies/options are present.